src/cgame/cg_gameinfo.cpp

```diff
@@ -31,12 +31,195 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+		if ( infos[ count ] )
+		{
+			strcpy( infos[ count ], info );
+			count++;
+		}
+	}
+
+	return count;
+}
+
+/*
+===============
+CG_LoadArenasFromFile
+===============
+*/
+static int CG_LoadArenasFromFile( const char *filename )
+{
+	std::error_code err;
+	std::string text = FS::PakPath::ReadFile( filename, err );
+	if ( err )
+	{
+		Log::Warn( "file not found: %s", filename );
+		return 0;
+	}
+
+	if ( text.size() >= MAX_ARENAS_TEXT )
+	{
+		Log::Warn( "file too large: %s is %i, max allowed is %i",
+			filename, text.size(), MAX_ARENAS_TEXT );
+
+		/* Tell it was found but not loaded properly,
+		it makes possible to not look for legacy file
+		that should not exist when the non-legacy file
+		actuallyexists, while not losing the information
+		and not returning true to not deceive futur users
+		of this function. */
+		return -1;
+	}
+
+	cg_numArenas += CG_ParseInfos( text.c_str(), MAX_ARENAS - cg_numArenas, &cg_arenaInfos[ cg_numArenas ] );
+
+	return 1;
+}
+
+/*
+===============
+CG_MapLoadNameCompare
+===============
+*/
+static int CG_MapLoadNameCompare( const void *a, const void *b )
+{
+	arenaInfo_t *A = ( arenaInfo_t * ) a;
+	arenaInfo_t *B = ( arenaInfo_t * ) b;
+
+	return Q_stricmp( A->name, B->name );
+}
+
 /*
 ===============
 CG_LoadArenas
 ===============
 */
-void CG_LoadArenas()
+void CG_LoadArenas( std::string mapname )
+{
+	cg_numArenas = 0;
+	rocketInfo.data.arenaCount = 0;
+
+	std::string arena_file = Str::Format( "meta/%s/%s.arena", mapname, mapname );
+
+	Log::Debug( "Loading arena file: %s", arena_file );
+
+	if ( CG_LoadArenasFromFile( arena_file.c_str() ) == 0 )
+	{
+		Log::Warn( "Arena file not found: %s", arena_file );
+
+		/* Legacy games may store more than one map metadata in a
+		single .arena file, the .arena file not having a basename
+		in common with any map, and store them in various folders
+		like scripts/ or maps/ folders. */
+
+		char dirlist[ 4096 ];
+		int numdirs = trap_FS_GetFileListRecursive( "", ".arena", dirlist, sizeof( dirlist ) );
+		char *dirptr = dirlist;
+
+		for ( int i = 0, dirlen = 0; i < numdirs; i++, dirptr += dirlen + 1 )
+	{
+		dirlen = strlen( dirptr );
+			Log::Warn( "Loading legacy arena file: %s", dirptr );
+			CG_LoadArenasFromFile( dirptr );
+		}
+	}
+
+	Log::Warn( S_SKIPNOTIFY "%i arenas parsed", cg_numArenas );
+
+	for ( int i = 0; i < cg_numArenas; i++ )
+	{
+		rocketInfo.data.arenaList[ rocketInfo.data.arenaCount ].name = BG_strdup( Info_ValueForKey( cg_arenaInfos[ i ], "map" ) );
+		rocketInfo.data.arenaList[ rocketInfo.data.arenaCount ].longName = BG_strdup( Info_ValueForKey( cg_arenaInfos[ i ], "longname" ) );
+		rocketInfo.data.arenaList[ rocketInfo.data.arenaCount ].author = BG_strdup( Info_ValueForKey( cg_arenaInfos[ i ], "author" ) );
+		rocketInfo.data.arenaCount++;
+
+		if ( rocketInfo.data.arenaCount >= MAX_ARENA_MAPS )
+		{
+			break;
+		}
+	}
+
+	qsort( rocketInfo.data.arenaList, rocketInfo.data.arenaCount, sizeof( mapInfo_t ), CG_MapLoadNameCompare );
```

Comment:
The sorting is useless since this list is only accessed by linear search

Reply:
Well, that was what the original code did, I can simplify it but all I did there was a revert.

Reply:
Remove the sorting, we don't use that list anywhere else anyway.